wonderous-book/lib.typ

```diff
@@ -70,26 +69,26 @@
     // The header always contains the book title on odd pages and
     // the chapter title on even pages, unless the page is one
     // that starts a chapter (the chapter title is obvious then).
-    header: locate(loc => {
+    header: context {
       // Are we on an odd page?
-      let i = counter(page).at(loc).first()
-      if calc.odd(i) {
+      if calc.odd(counter(page).get().first()) {
         return text(0.95em, smallcaps(title))
       }
 
       // Are we on a page that starts a chapter? (We also check
       // the previous page because some headings contain pagebreaks.)
-      let all = query(heading, loc)
+      let i = here().page()
+      let all = query(heading)
       if all.any(it => it.location().page() in (i - 1, i)) {
         return
       }
 
       // Find the heading of the section we are currently in.
-      let before = query(selector(heading).before(loc), loc)
+      let before = query(selector(heading).before(here()))
       if before != () {
         align(right, text(0.95em, smallcaps(before.last().body)))
       }
-    }),
```

Comment:
Can the repetition of `text(0.95em)` be avoided on lines 89 and 75; instead add set text(0.95em) before context ends (i.e. before line 92)?

Reply:
I'm afraid that wouldn't work as `return` causes styles to be ignored. We could consider introducing a variable if you prefer.

Reply:
> I'm afraid that wouldn't work as `return` causes styles to be ignored. We could consider introducing a variable if you prefer.

Thank you for considering my comment; much appreciate. No, I'd follow your kind advice, and leave it as it is.